awx/main/dispatch/pool.py

```diff
@@ -0,0 +1,359 @@
+import logging
```

Comment:
The code in this module implements the worker pool; it includes autoscaling code that forks new worker processes, checks their health and replaces ones that exit/die, and spins down workers that are idle.

```python
    class MessagePrinter(awx.main.dispatch.worker.BaseWorker):

         def perform_work(self, body):
             print body

     pool = WorkerPool(min_workers=4)  # spawn four worker processes
     pool.init_workers(MessagePrint().work_loop)
     pool.write(
         0,  # preferred worker 0
         'Hello, World!'
     )
```

As AMQP messages for the queues we care about come in, we call `pool.write()`, and the messages are distributed amongst the worker process in the pool.